README.md

```diff
@@ -1,17 +1,130 @@
+
+```ipldsch
+type UCAN struct {
+  version String
+
+  issuer DID
+  audience DID
+  signature Signature
+
+  capabilities [Capability]
+  proofs [&UCAN]
+  expiration Int
+
+  facts [Fact]
+  nonce optional String
+  notBefore optional Int
+} representation map {
+  field facts default []
+  field proofs default []
+}
+```
+
+## 2.1 DID
+
+DIDs MUST be encoded as [DID](https://www.w3.org/TR/did-core/)s. The [`did:key` method](https://w3c-ccg.github.io/did-method-key/) is RECOMMENDED as it is self-contained.
+
+``` ipldsch
+type DID = String
+```
+
+## 2.2 Capability
+
+``` ipldsch
+type Capability struct {
+  with Resource
+  can Ability
+  extension optional Extension
+} representation map
+```
+
+### 2.2.1 Resource
+
+The resource pointer MUST be formatted as a [URI](https://www.rfc-editor.org/rfc/rfc3986).
+
+``` ipldsch
+type Resource = String
+```
+
+### 2.2.2 Ability
+
+Abilities MUST be lowercase, and MUST be namespaced with `/` delimeters. Abilities SHOULD have at least one path segment.
+
+``` ipldsch
+type Ability = String
+```
+
+### 2.2.3 Extensions
+
+The extended capability field is OPTIONAL. When present, it MUST contain any additional domain specific details and/or restrictions of the capability. 
+
+``` ipldsch
+type Extension = { String: Any }
+```
+
+## 2.3 Fact
+
+Facts MUST be structured as maps with string keys.
+
+``` ipldsch
+type Fact { String: Any }
+```
+
+## 2.4 Proof
+
+As of UCAN 0.9, all proofs MUST be given as links (CIDs). Note that UCANs MAY be inlined via the [identity multihash (`0x00`)](https://github.com/multiformats/multicodec/blob/master/table.csv#L2) CID.
+
+``` ipldsch
+&UCAN
+```
+
+## 2.5 Signature
+
+The signature MUST be computed by first encoding it as a [canonical JWT](#3-jwt-canonicalization), and then signed with the issuer's private key.
+
+``` ipldsch
+type Signature = Bytes
+```
+
+# 3 JWT Canonicalization
+
+Per the core UCAN spec, all implementations MUST support JWT encoding. This provides a common representation that all implementations can understand. JWT canonicalization allows for an IPLD UCAN to be expressed as a JWT, retain the JWT signature scheme, and so on for compatibility, while retaining the ability to translate into other formats for storage or transmission among IPLD-enabled peers.
+
+To canonicalize an IPLD UCAN to JWT, the JSON segments MUST be encoded per [`dag-json`](https://ipld.io/specs/codecs/dag-json/spec/), encoded as unpadded [base64url](https://datatracker.ietf.org/doc/html/rfc4648#section-5), and joined with `.`s.
+
+UCAN canonicalization is signalled by CID. If no canonicalization is used, the CID MUST use the [raw multicodec](https://github.com/multiformats/multicodec/blob/master/table.csv#L39). Canonicalized UCANs that wish to signal this encoding MUST use [any other CID codec](https://github.com/multiformats/multicodec/blob/master/table.csv), including but not limited to `dag-json` and `dag-cbor`.
```

Comment:
If UCAN canonicalization is signalled by CID, how do you signal a UCAN that you send directly, let's say as a base64-encoded header directly without a CID in between?
I guess you could use the identity hash to do the encoding + codec signalling at the same time :thinking: 

It seems the difference now is that the UCAN itself doesn't contain the information about its canonicalization itself, only the *pointer to* the UCAN has (the CID).

Shouldn't be a problem though if you're packaging up UCANs in CAR files which contain the root UCAN's CID.

Reply:
> If UCAN canonicalization is signalled by CID, how do you signal a UCAN that you send directly, let's say as a base64-encoded header directly without a CID in between?
> I guess you could use the identity hash to do the encoding + codec signalling at the same time 🤔

I don't know if I fully understand what you're saying, but you could embed / inline proofs with identity hash indeed. That is precisely what we're recommending in dag-ucan implementation
https://github.com/ipld/js-dag-ucan#embedding-proofs

Reply:
@Gozala he's saying that you if we send a UCAN by itself without its CID, we won't know if it canonicalized because there's no signal in the UCAN itself. In a previous draft, there was a flag in the header. I removed it because:

1. It's one more thing that can break
2. We can always determine if it's canonicalized by round-tripping it through an IPLD codec

Reply:
> We can always determine if it's canonicalized by round-tripping it through an IPLD codec

Yeah that is precisely how dag-ucan goes about it.